homeassistant/components/tcp.py

```diff
@@ -0,0 +1,23 @@
+"""
+homeassistant.components.tcp
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+A generic TCP socket component.
+"""
+DOMAIN = "tcp"
+
+CONF_PORT = "port"
+CONF_TIMEOUT = "timeout"
+CONF_PAYLOAD = "payload"
+CONF_UNIT = "unit"
+CONF_VALUE_REGEX = "value_regex"
+CONF_VALUE_TEMPLATE = "value_template"
+CONF_VALUE_ON = "value_on"
+CONF_BUFFER_SIZE = "buffer_size"
+
+DEFAULT_TIMEOUT = 10
+DEFAULT_BUFFER_SIZE = 1024
+
+
+def setup(hass, config):
+    """ Nothing to do! """
```

Comment:
If there is nothing to do then let's not have it exist. Like In my prior comment - just have binary sensor depend on sensor.